### PR TITLE
Convert writeAccessFactory to a serializable suppliar

### PR DESCRIPTION
**Pull request by valiparsa**

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/direct_es/ElasticSearchWriteAccessDirect.java [hunk 1]
```diff
@@ -97,7 +98,6 @@ public class ElasticSearchWriteAccessDirect extends ElasticSearchWriteAccess {
     public ElasticSearchWriteAccessDirect() throws URISyntaxException {
         ElasticSearchConfiguration config = ElasticSearchGlobalConfiguration.get().getElasticSearch();
         if(config != null) {
-            config.readResolve();
```
Comment (dimitrij-afonitschkin) [inline src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/direct_es/ElasticSearchWriteAccessDirect.java:100]: Why is the line removed?
It initializes a lot of properties not just splitMessagesLongerThan.
Is function readResolve() used somewhere else? If not, it should be deleted.

Comment (valiparsa) [reply src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/direct_es/ElasticSearchWriteAccessDirect.java:100]: Actually it was initially not called. [I changed it here](https://github.com/SAP/elasticsearch-logs-plugin/pull/87/files#diff-4d0e15af1f1c1a7a36265c7c8695d8f864f146918c92a9a13c86d57d0a29c346) to be called. But I guess readObject() and readResolve() methods are such methods [which are being called automatically during serialization](https://www.baeldung.com/java-serialization-readobject-vs-readresolve) right? That is why I removed them as part of this PR.

Comment (dimitrij-afonitschkin) [reply src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/direct_es/ElasticSearchWriteAccessDirect.java:100]: I have learned something new. 
It looks like readResolve() is called internally during serialization.

**Review by dimitrij-afonitschkin — APPROVED** [review]
Looks good for me.

File: src/main/resources/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter/config.jelly [hunk 2]
```diff
@@ -13,30 +13,39 @@
...
+  </f:entry>
 
+  <f:entry field="bufferChunkRetentionTimeMillis" title="Buffer Retention time (ms)">
+    <f:number clazz="number" min="0" default="1000" max="2147483647"/>
+  </f:entry>
+
+  <f:entry field="maxBufferSize" title="Max Buffer size (bytes)">
+    <f:number clazz="number" min="65535" default="10485760" max="2147483647"/>
+  </f:entry>
 </j:jelly>
```
Comment (valiparsa) [inline src/main/resources/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter/config.jelly:51]:
```suggestion
</j:jelly>

```

File: src/main/resources/io/jenkins/plugins/pipeline_elasticsearch_logs/ElasticSearchConfiguration/config.jelly [hunk 3]
```diff
@@ -26,8 +26,8 @@
     <f:entry >
       <f:dropdownDescriptorSelector title="RunID Provider" field="runIdProvider"/>
     </f:entry>
-    <f:entry field="connectionTimeoutMillis" title="Connection Timeout (ms)">
-      <f:number clazz="number" min="0" max="2147483647" default="10000"/>
+    <f:entry field="splitMessagesLongerThan" title="Split messages which are longer than this size (byte)">
+      <f:number clazz="number" min="0" max="2147483647" default="20000"/>
     </f:entry>
   </f:advanced>
 </j:jelly>
```
Comment (valiparsa) [inline src/main/resources/io/jenkins/plugins/pipeline_elasticsearch_logs/ElasticSearchConfiguration/config.jelly:33]:
```suggestion
</j:jelly>

```

**Review by romanisb — APPROVED** [review]

**Pull request merged**